### ISSUE TITLE
feat: support Type Declerations

### ISSUE THREAD

**Issue by econneely** [module-federation/vite#112]
Does this support the hotreloading of types mentioned here and if so is there any examples of how to implement it https://module-federation.io/guide/basic/type-prompt.html

Does this have something to do with it not working https://github.com/module-federation/vite/blob/main/src/plugins/pluginMFManifest.ts#L376-L377

`react refresh:6 [ dynamic-remote-type-hints-plugin ] err: [object Event]` this is also one of the errors on the host side and the remote is using webpack and its manifest has a reference to the .d.ts and zip

**Comment by gioboa** [comment]
Hi @econneely, can you open a PR with a failing scenario in [this example](https://github.com/module-federation/vite/tree/main/examples/vite-webpack-rspack) pls? With your help we will solve the problem sooner. Thanks

**Comment by econneely** [comment]
@gioboa that example fails on the host example with this error `Failed to resolve entry for package "@module-federation/vite". The package may have incorrect main/module/exports specified in its package.json. [plugin externalize-deps`

**Comment by econneely** [comment]
I ran pnpm install then pnpm run multi-example

**Comment by gioboa** [comment]
> I ran pnpm install then pnpm run multi-example

I cloned the repo from scratch and you need to run `pnpm run build`  after the install

**Comment by econneely** [comment]
Update the readme with tha that fixed it will get the pr up asap

**Comment by gioboa** [comment]
> Update the readme with that, it fixed it will get the pr up asap

Fixed, thanks

**Comment by econneely** [comment]
Hi any timeframe on this want to know for planning some work with a Vite project thanks

**Comment by gioboa** [comment]
I'm on it, I need few days to figure out a solution.

**Comment by econneely** [comment]
Thanks much appreciated

**Comment by gioboa** [comment]
The idea is to use the [@module-federation/dts-plugin]( https://www.npmjs.com/package/@module-federation/dts-plugin) because it's always up to date. I'm trying to feed it with a generic compiler. Let's see

**Comment by Haprog** [comment]
The issue title is "Hotreloading Type Declerations" but isn't it so that not even loading once is supported yet (even without hot reload)? Maybe the issue should be renamed to "Support loading Type Declarations (DTS)" or similar?

I would be happy to initially have even a manual way to generate the `@mf-types` directory (either with build plugin or even a separate command I can run) and hot relaoding could be added later (if that is more complicated than implementing loading once).

**Comment by tduforet** [comment]
Does a branch already exist for this issue?

**Comment by gioboa** [comment]
Not really, I tried something in my local env without pushing anything. You can start form the main branch.
